Sorting/BucketSort.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+    Bucket sort
+    ----------
+    Bucket sort, or bin sort, is a sorting algorithm that works by distributing the elements of an array into a number of buckets. 
+    Each bucket is then sorted individually, 
+    either using a different sorting algorithm, 
+    or by recursively applying the bucket sorting algorithm.
+
+    Time complexity
+    ---------------
+    O(N), where N is the number of elements.
+
+    Space complexity
+    ----------------
+    O(N), where N is the number of elements.
+    
+*/
+
+#include <iostream>
+#include <vector>
+
+#include "SortingUtils.hpp"
+
+using namespace std;
+
+void bucket_sort (vector<int>& values, const int order, const bool to_show_state)
+{
+    size_t i, j;
+    int m = 101;
```

Comment:
Here m is the maximum range of number .
when m = 101 , We cannot use any number more than 101.

I can do two things now -
     1. Make m = 999999999999 (some very large value) and declare it globally.
     2. I can declare int m   globally and get its value from the user by adding cin>>m at the first   line of int main()
Which one you prefer ? let me know